### PR TITLE
Remove Sinatra dependency, provide the minimum possible Sinatra compatibility layer

### PR DESCRIPTION
**Pull request by julik**

### CODE REVIEW

**Comment by julik** [comment]
@dennisvdvliet can you lend a hand / take this over? This holds us from upgrading to Rails 5 actually.

**Comment by julik** [comment]
❤️ ❤️ ❤️

**Comment by dennisvdvliet** [comment]
I hope I have some time today to test this new version with our own frontend repo to see what we break.

File: spec/apiculture_spec.rb [hunk 1]
```diff
@@ -10,13 +10,11 @@ def app
...
     it 'allows all the standard Siantra DSL to go through without modifications' do
-      @app_class = Class.new(Sinatra::Base) do
-        settings.show_exceptions = false
-        settings.raise_errors = true
+      @app_class = Class.new(Apiculture::App) do
         extend Apiculture
       
         post '/things/*' do
-          return params.inspect
+          params.inspect
```
Comment (dennisvdvliet) [inline spec/apiculture_spec.rb]: @julik Do you think we should make this work again?

**Comment by dennisvdvliet** [comment]
Works on the following wt apps

- [x] frontend
- [x] nu_shorturls
- [ ] air_traffic_control

**Comment by dennisvdvliet** [comment]
@julik Shall we merge and release this? Maybe try it out on air_traffic_control or shorturl before we try frontend

**Comment by julik** [comment]
Yes, squash + merge + major version bump please. I'm busy with something atm 🎉

**Pull request merged**